contracts/Datastore.sol

```diff
@@ -134,16 +139,48 @@ contract Datastore is AragonApp {
         writeAccess = hasWriteAccess(_fileId, _caller);
     }
 
-
+    /**
+     * @notice Returns the encryption key for file with `_fileId`
+     * @param _fileId File Id 
+     */
+    function getFileEncryptionKey(uint _fileId) external view returns(string) {
+        if (hasReadAccess(_fileId, msg.sender)) {
+            FileLibrary.File storage file = fileList.files[_fileId];
+            return file.cryptoKey;
+        }
+        return "0";
+    } 
 
     /**
-     * @notice Delete file with Id `_fileId`
+     * @notice Set file `_fileId` as deleted or not.
      * @param _fileId File Id
+     * @param _isDeleted Is file deleted or not
+     * @param _deletePermanently If true, will delete file permanently
      */
-    function deleteFile(uint _fileId) public onlyFileOwner(_fileId) {
-        fileList.deleteFile(_fileId);
+    function deleteFile(uint _fileId, bool _isDeleted, bool _deletePermanently) public onlyFileOwner(_fileId) {
+        if (_isDeleted && _deletePermanently) {
+            fileList.permanentlyDeleteFile(_fileId);
+            emit DeleteFilePermanently(msg.sender);            
```

Comment:
Shouldn't these delete events include the `fileId`?

Reply:
Yeah we started doing a clean up of our events (#83)

Reply:
Hm, I don't think I get it. In that PR it seems you are even removing more fields, up to the point of leaving events empty. How are you planning to use the events. How will you know which objects the events are related to?

Reply:
Yes sorry Bigen  I didn't explain myself very well. We started doing a big clean up of events because the majority of the parameters were not really useful and we didn't used them. So to optimize our bytecode size, we removed every params and a few events we thought were redundant. Now that the clean up is done in #83 we're supposed to start adding more params like fileId and groupId and more that we think can be useful for the users.

Reply:
Ok, makes sense. I'll have another look at it when events are fulfilled.

Reply:
We started adding event params in this pull request: #86 We are still debating on what to add next. Maybe for `FileChange` and `SettingsChange` we will add the type of change.